modules/optimization/doc/content/source/reporters/OptimizationData.md

```diff
@@ -14,15 +14,27 @@
 
+If the [!param](/Reporters/OptimizationData/objective_name) is set by the user
+`OptimizationData` will create and store the L2 Norm of the misfit
+values divided by two. This can be useful for running an inverse optimization problem with the
+[GeneralOptimization.md] reporter.
+
 ## Example Input File Syntax
 
-`OptimizationData` is typically used in the forward and homogeneous optimization sub-applications to sample the solution for `simulation_values` and calculating `misfit_values`. The measurement data can either be specified directly or (more commonly) transferred from the optimization main application, while the simulation and misfit values are transferred to the main application. It is also typically used in the adjoint/gradient sub-application to hold the misfit values to apply the [ReporterPointSource.md] Dirac kernel.
+`OptimizationData` is typically used in the forward and homogeneous optimization
+sub-applications to sample the solution for `simulation_values`, calculating
+`misfit_values`, and storing the objective value provided by `objective_name`.
+The measurement data can either be specified directly or
+(more commonly) transferred from the optimization main application, while the
+simulation and misfit values are transferred to the main application. It is also
+typically used in the adjoint/gradient sub-application to hold the misfit values
+to apply the [ReporterPointSource.md] Dirac kernel.
 
-The following code blocks demonstrate this utilization of `OptimizationData`. The [OptimizationReporter.md] in the main application reads the measurement locations and values. A [MultiAppReporterTransfer.md] transfers the measurement data to the forward, adjoint/gradient, and homogeneous application within the `OptimizationData` reporter declared in those inputs. The forward and homogeneous applications then sample the specified vector specified in [!param](/Reporters/OptimizationData/variable). The main application then receives the simulation data using a [MultiAppReporterTransfer.md], which [OptimizationReporter.md] uses to calculate the misfit and transfer to the adjoint/gradient sub-application, which uses that data to apply the [ReporterPointSource.md] kernel.
+The following code blocks demonstrate this utilization of `OptimizationData`. The [OptimizationData.md] in the main application reads the measurement locations and values. A [MultiAppReporterTransfer.md] transfers the measurement data to the forward, adjoint/gradient, and homogeneous application within the `OptimizationData` reporter declared in those inputs. The forward and homogeneous applications then sample the specified vector specified in [!param](/Reporters/OptimizationData/variable). The main application then receives the simulation data using a [MultiAppReporterTransfer.md], which [OptimizationReporter.md] uses to calculate the misfit and transfer to the adjoint/gradient sub-application, which uses that data to apply the [ReporterPointSource.md] kernel.
```

Comment:
Are we still using  [OptimizationReporter.md] in this problem?   Or is this just the name of the block or action?